### PR TITLE
Integrate swift lint

### PR DESCRIPTION
**Pull request by TStrawberry**
## Overview
Add lint for iOS

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## In scope
- Integrate SwiftLint
- Add lane in Fastfile to lint and autocorrect Swift code
- Add pre-push to check lint

## Check list
1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?
3. [x] Have you update the wiki of your change?

## Reference
- Card: https://trello.com/c/GmXtv5sM

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -37,6 +37,30 @@ platform :ios do
     run_tests(scheme: "ARK-iOS")
   end
 
+  desc "Runs all the static check"
+  lane :check do
+    swiftlint(
```
Comment (Tony---Zhang) [inline fastlane/Fastfile]: 这里直接调用lint_correct就可以了

**Review by Tony---Zhang — COMMENTED** [review]
Fix fastlane issue

File: fastlane/Fastfile (hunk 1 above)
Comment (monkey-tw) [reply fastlane/Fastfile]: 这是只是check，并没有自动修正。所以不能调用下面定义好的lint_correct

File: fastlane/Fastfile [hunk 2]
```diff
@@ -37,6 +37,29 @@ platform :ios do
     run_tests(scheme: "ARK-iOS")
   end
 
+  desc "Runs all the static check"
+  lane :check do
+    lint_base({ :mode => :lint })
+  end
+
+  desc "Autocorrect all swift files"
+  lane :lint_correct do
```
Comment (WeidongGu) [inline fastlane/Fastfile:46]: 这个是不是也需要在 readme 里添加描述呀

**Review by WeidongGu — APPROVED** [review]

**Review by pujiaxin33 — APPROVED** [review]

File: fastlane/Fastfile (hunk 2 above)
Comment (Tony---Zhang) [reply fastlane/Fastfile:46]: 我们有个WIKI，里面有说

**Review by Tony---Zhang — APPROVED** [review]

**Pull request merged**